src/test/test_cylinder_faceting.cc

```diff
@@ -0,0 +1,70 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include "brep_faceter.hh"
+#include "BRepTools.hxx"
+#include "moab/GeomQueryTool.hpp"
+#include "moab/GeomTopoTool.hpp"
+#include "BRepPrimAPI_MakeCylinder.hxx"
+
+TEST_CASE("Faceting cylinder and writing to MOAB", "[cylinder_faceting]") {
+
+  FacetingTolerance facet_tol(1.e-3);
+
+  MBTool mbtool;
+  mbtool.set_faceting_tol_tag(facet_tol.tolerance);
+
+  const char *brep_path = "cylinder.brep";
+  const char *output_path = "cylinder_test_output.h5m";
+  
+  float radius = 2;
+  float height = 10;
+  gp_Ax2 anAxis;
+  anAxis.SetLocation(gp_Pnt(0.0, 0, 0.0));
+  TopoDS_Shape shape = BRepPrimAPI_MakeCylinder(anAxis, radius, height).Shape();
+
+  BRepTools::Write(shape, brep_path);
+
+  REQUIRE(!shape.IsNull());
+
+  MaterialsMap materials_map;
+  std::uint64_t uniqueID = calculate_unique_id(shape);
+  materials_map[uniqueID] = "testmat";
+
+  sew_and_facet(shape, facet_tol, mbtool, materials_map);
+
+  std::vector<moab::EntityHandle> triangles = mbtool.get_entities_by_dimension(0, 2, true);
+
+  // This is a check if something has changed - I haven't independantly checked this value:
+  CHECK(triangles.size() == 322);
+
+  // 2 vertices + 3 edges + 3 faces + 1 volume + 1 material group = 10
```

Comment:
Although this cylinder faceting test is passing, I'm far from convinced that there are sufficient checks, or that the output file is correct.  The comment `// 2 vertices + 3 edges + 3 faces + 1 volume + 1 material group = 10` appears to be wrong - I think there are other reasons for the 10 meshsets (which include the gather set).